Rick-and-Morty/Rick And Morty/Views/CharacterCard.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  CharacterTileView.swift
+//  Rick And Morty
+//
+//  Created by Scottie Gray on 2021-08-10.
+//  Copyright © 2021 Novoda. All rights reserved.
+//
+
+import SwiftUI
+import Foundation
+
+struct CharacterCardState {
+    var id: Int
+    var name: String
+    var image: UIImage
+    var isAlive: Bool
+    var species: String
+    var lastLocation: String
+    var firstEpisode: String
+}
+
+struct CharacterCard: View {
+    var characterCardState: CharacterCardState
+    
+    private let cornerRadius: CGFloat = 10
+    private let noSpacing: CGFloat = 0
+    private let VSpacing: CGFloat = 10
+    
+    var body: some View {
+        HStack(alignment: .top) {
+            Image(uiImage: characterCardState.image)
+                .resizable()
+                .aspectRatio(contentMode: .fit)
+            
+            VStack(alignment: .leading, spacing: VSpacing) {
+                VStack(alignment: .leading, spacing: noSpacing) {
+                    Text(characterCardState.name)
+                        .font(.title)
+                        .fontWeight(.black)
+                    HStack {
+                        Circle()
+                            .foregroundColor(characterCardState.isAlive ? .green : .red)
+                            .frame(width: 10, height: 10)
+                        Text(characterCardState.isAlive ? "Alive - \(characterCardState.species)" : "Dead - \(characterCardState.species)")
+                    }
+                }
+                
+                VStack(alignment: .leading, spacing: noSpacing) {
+                    Text("Last known location:")
+                        .font(.caption)
+                        .foregroundColor(.secondary)
+                    Text(characterCardState.lastLocation)
+                }
+                
+                VStack(alignment: .leading, spacing: noSpacing) {
+                    Text("First seen in:")
+                        .font(.caption)
+                        .foregroundColor(.secondary)
+                    Text(characterCardState.firstEpisode)
```

Comment:
The duplication here could be extracted to a function. I havent played with SwiftUI to know enough but roughly along the lines of:

```
func characterDetailsVStack(pretext: String, details: String) -> VStack {
    return VStack(alignment: .leading, spacing: noSpacing) {
                    Text(pretext)
                        .font(.caption)
                        .foregroundColor(.secondary)
                    Text(details)
}
```
then called:

```
    characterDetailsVStack(pretext: "Last known location:", details: characterCardState.lastLocation)
    characterDetailsVStack(pretext: "First seen in:", details: characterCardState.firstEpisode)

Reply:
Id maybe even consider putting those strings as constants too, or prep them for localisation 🤷